flowlogs_reader/flowlogs_reader.py

```diff
@@ -158,71 +158,29 @@ def next(self):
         # For Python 2 compatibility
         return self.__next__()
 
-    def _get_log_stream_data(self):
-        # Loops through the pages of logs streams, returning the list of
-        # available streams.
+    def _read_streams(self):
         next_token = None
-        kwargs = {'logGroupName': self.log_group_name}
-        ret = []
-
-        while True:
-            if next_token:
-                kwargs['nextToken'] = next_token
-
-            response = self.logs_client.describe_log_streams(**kwargs)
-            ret.extend(response['logStreams'])
-
-            next_token = response.get('nextToken')
-            if not next_token:
-                break
-
-        return ret
-
-    def _filter_log_streams(self, log_stream_data):
-        # Filters out log streams that were ingested before the given
-        # start time or after the given end time.
-        ret = []
-        for log_stream in log_stream_data:
-            last_ingest_time = log_stream.get('lastIngestionTime')
-            if not last_ingest_time:
-                continue
-
-            if self.start_ms <= last_ingest_time < self.end_ms:
-                ret.append(log_stream['logStreamName'])
-
-        return ret
-
-    def _read_stream(self, stream_name):
-        # Loops through the pages of the log stream with the given
-        # `stream_name`, yielding the events.
-        forward_token = None
         kwargs = {
             'logGroupName': self.log_group_name,
-            'logStreamName': stream_name,
             'startTime': self.start_ms,
             'endTime': self.end_ms,
-            'startFromHead': True,
+            'interleaved': True,
         }
 
         while True:
-            if forward_token:
-                kwargs['nextToken'] = forward_token
+            if next_token:
+                kwargs['nextToken'] = next_token
 
-            response = self.logs_client.get_log_events(**kwargs)
+            response = self.logs_client.filter_log_events(**kwargs)
             for event in response['events']:
                 yield event
 
-            next_forward_token = response['nextForwardToken']
-            if next_forward_token == forward_token:
+            next_token = response.get('nextToken')
+            if not next_token:
```

Comment:
Seems like you could tweak this to be:
```python
next_token = response.get('nextToken')
if next_token:
    kwargs['nextToken'] = next_token
else:
    break
```

And remove the variable from line 162?